### ISSUE TITLE
[A11Y]: TextArea invalid text is not accessible

### ISSUE THREAD

**Issue by emmafull** [IBM/carbon-components-react#1529]
## Detailed description

> Describe in detail the issue you're having.

TextArea invalid text cannot be read by the screenreader JAWS (Job Access With Speech).

> Is this issue related to a specific component?

TextArea

> What version of the Carbon Design System are you using?

"carbon-components": "9.26.0"
"carbon-components-react": "6.39.3"

> What offering/product do you work on? Any pressing ship or release dates we should be aware of?

IBM Cloud Internet Services

## Additional information
===== ENVIRONMENT =====
IBM CI 162 Checkpoint Violation: 3.3.1 Error Identification:
IBM Guidance: https://www.ibm.com/able/guidelines/ci162/error_id.html
Test Tool Type: JAWS

===== DEFECT DESCRIPTION =====
If an input error is automatically detected, the item that is in error is identified and the error is described to the user in text.

The purpose of this checkpoint is to ensure that users are aware that an input error has occurred and can determine what is wrong. The error message should be as specific and detectable as possible.

**I will submit a PR.**
